.github/workflows/validator.yml

```diff
@@ -34,68 +34,48 @@ jobs:
-
-    - name: git config
+    - name: Install spine-generic
       run: |
-        # this is needed by git-annex so that it can write to the local git-annex branch
-        # the tracking information about the local copy
-        git config --global user.name "gh-actions"
-        git config --global user.email "gh-actions"
-        # but actually, disable those writes. We don't need them because we're throwing away this copy immediately.
-        # this might make the previous two lines unnecessary, but it's safer just to leave them both in
-        git config annex.alwayscommit false
-
-    - name: Download dataset
-      run: |
-        git fetch --depth=1 origin git-annex:git-annex # actions/checkout@v4 does a shallow checkout, so it is missing this important branch
-        git annex init
-        git annex get -J8
+        pip install spinegeneric@git+https://github.com/spine-generic/spine-generic.git@v2.9
 
-    - name: Checking BIDS compliance
-      if: always()
+    - name: Run bids-validator
+      # If the explicit whitelist of allowed environment variables gets too annoying to update,
+      # we can loosen it to just a blanket `--allow-env`. This will let bids-validator and all of its
+      # dependencies access any credentials that github CI exposes in environment variables, but at
+      # least we're not giving it any network access, so exfiltrating them would require more effort.
       run: |
-        bids-validator --verbose ./ | tee bids.txt
-        printf "\nShort summary:\n\n"
-        # FIXME: This should be replaced with bids-validator settings: A) throw error on warning or B) warning ignores
-        ! grep "\[WARN\]\|\[ERR\]" bids.txt  # ! -> Reverse error code so "found == error thrown"
+        deno run --no-prompt --allow-env=BIDS_SCHEMA,FORCE_COLOR,IGNORE_TEST_WIN32 --allow-read=. jsr:@bids/validator . || echo '::warning title=bids-validator::Errors found by bids-validator'
 
-    - name: Checking acquisition parameters
-      if: always()
+    - name: Run acquisition parameter validator
       run: |
-        sg_params_checker -path-in ./ | tee params.txt
-        # FIXME: Checking the output like this should be replaced with SG throwing actual errors
-        ! grep "WARNING" params.txt  # ! -> Reverse error code so "found == error thrown"
+        sg_params_checker -path-in . || echo '::warning title=bids-validator::Errors found by sg_params_checker'
 
-    - name: Checking data consistency
-      if: always()
+    - name: Run data consistency validator
       run: |
-        sg_check_data_consistency -path-in ./ | tee consistency.txt
-        # FIXME: Checking the output like this should be replaced with SG throwing actual errors
-        ! grep "Warning\|Missing" consistency.txt
-        grep "all good" consistency.txt
+        sg_check_data_consistency -path-in . || echo '::warning title=bids-validator::Errors found by sg_check_data_consistency'
```

Comment:
This syntax can be finicky sometimes, but since we're not substituting any data into the echo string, and only using the `title=` key, this looks good to me!

(I had to wrangle with it a bit for [this warning](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/master/.github/workflows/test-batch-processing.yml#L163), but it's much simpler here.)

Reply:
Exactly! Haha this is why I had a few force-pushes to this PR's branch near the end.